index.js

```diff
@@ -0,0 +1,25 @@
+module.exports = (timeoutMs, exitCode) => ({
+  name: "startChecker",
+  starting(broker) {
+    this.exitCode = 0;
+    this.timeoutID = setTimeout(() => {
+      broker.logger.error(
+        "some services did not start in time, stopping broker"
+      );
+      this.exitCode = exitCode;
+      broker.stop();
+    }, timeoutMs);
+  },
+  started(broker) {
+    clearTimeout(this.timeoutID);
+  },
+  stopped(broker) {
+    if (process.ENV_NAME && process.ENV_NAME === "TEST_ENV") {
```

Comment:
The best way to handle this would be to accept an `onStartTimeout` callback as an option instead of `exitCode`.
The default callback would do `process.exit(1)`.
If anyone wants to change the exit code, they just have to overload the callback.
For testing purposes, the callback would tell the current test that the timeout has been reached.